activerecord/lib/active_record/querying.rb

```diff
@@ -69,6 +69,11 @@ def _load_from_sql(result_set, &block) # :nodoc:
         column_types = column_types.reject { |k, _| attribute_types.key?(k) }
       end
 
+      duplicate_column_names = result_set.columns.tally.select { |_, v| v > 1 }.keys
+      if duplicate_column_names.any?
+        logger.warn "Duplicate column names: #{duplicate_column_names.join(', ')}"
```

Comment:
This message isn't actionable. And the SQL can be executed very far away form the code that generated it, so we need to include which SQL has the duplicated names. In fact, I'm not sure a warning is the right thing to do here. Is there any case were this behavior is intended and not a mistake?

Reply:
This behavior _might_ be intended (in terms of user who can explicitly name two columns the same), but results in a model that might have odd properties. I'd say the most common problem is when you have multiple identity columns, since you can inadvertently fetch a record, update it and save it with a wrong `id`, overwriting some data.

I've stumbled upon this behaviour in client's app code that's why I'm interested in at least logging this case. Perhaps we can just add a warning for multiple identity columns? 

Also - any suggestions for better error message?